### PR TITLE
Add tests for exponential extrapolation

### PR DESCRIPTION
**Pull request by natestemen**
**Context:** The last PR ([first](https://github.com/PennyLaneAI/catalyst/pull/806), [second](https://github.com/PennyLaneAI/pennylane/pull/5972)) to add exponential extrapolation capabilities to catalyst.

**Description of the Change:** This PR ensures that `pennylane.transforms.exponential_extrapolate` works inside catalyst's `mitigate_with_zne` function.

**Benefits:** ZNE has increased functionality.

**Possible Drawbacks:** As part of the testing, I've removed 0.1 and 0.2 from the list of values being used to create different circuits. This is because exponential extrapolation is not necessarily as stable as polynomial fitting. For example, when extrapolating near constant values exponential extrapolation can struggle due to the fact that a linear fit is first performed to understand the "direction" (or `sign`) of the exponential. If the slope is positive, the data is flipped to fit something that looks like $A\mathrm{e}^{-Bx} + C$ where $B\in\mathbb{R}_{> 0}$.

An example of this happening can be seen here.
```py
>>> exponential_extrapolate([1,2,3], [0.3894, 0.3894183, 0.38941])
-1.0000000000000059e-06
>>> richardson_extrapolate([1,2,3], [0.3894, 0.3894183, 0.38941])
0.3893551000000072
```

If we want to ensure to continue testing values 0.1 and 0.2 for polynomial extrapolation, we can create separate tests. Let me know what would be best.

**Related GitHub Issues:**

fixes https://github.com/PennyLaneAI/catalyst/issues/754

### CODE REVIEW

**Comment by natestemen** [comment]
just pinging @rmoyard / @dime10 / @josh146 in case y'all didn't see the discord message.

File: frontend/test/pytest/test_mitigation.py [hunk 1]
```diff
@@ -18,15 +18,17 @@
 import numpy as np
 import pennylane as qml
 import pytest
+from pennylane.transforms import exponential_extrapolate
 
 import catalyst
 from catalyst.api_extensions.error_mitigation import polynomial_extrapolation
```
Comment (cosenal) [inline frontend/test/pytest/test_mitigation.py:24]: Did you figure out if it makes sense to create a similar wrapper for `exponential_extrapolation` as well?

File: frontend/test/pytest/test_mitigation.py [hunk 2]
```diff
@@ -308,5 +315,30 @@ def mitigated_qnode():
     assert np.allclose(mitigated_qnode(), circuit())
 
 
+def test_exponential_extrap_with_kwargs():
```
Comment (cosenal) [inline frontend/test/pytest/test_mitigation.py:318]: micro-nit: may as well just write the full word 😀

`extrap` sounds like a musical genre that has waned in popularity.

**Review by cosenal — APPROVED** [review]
Lgtm

**Review by rmoyard — COMMENTED** [review]
@natestemen could keep all the values but add a skip in the test directly for those values only for exponential? With the skip add a comment about exponential not being stable.

**Comment by rmoyard** [comment]
Closes https://github.com/PennyLaneAI/catalyst/issues/754

File: frontend/test/pytest/test_mitigation.py (hunk 2 above)
Comment (natestemen) [reply frontend/test/pytest/test_mitigation.py]: fixed (along with a few other instances) here: https://github.com/PennyLaneAI/catalyst/pull/953/commits/eea7d4d27408b4351525acd4ea39310841bb9e46

**Review by rmoyard — APPROVED** [review]
Thanks :100:

File: frontend/test/pytest/test_mitigation.py (hunk 1 above)
Comment (rmoyard) [reply frontend/test/pytest/test_mitigation.py:24]: Needs to be resolved before merging @natestemen

**Comment by natestemen** [comment]
@rmoyard just added a snippet to the changelog, and added a missing docstring to satisfy CI. Feel free to take another look, or merge!

File: frontend/test/pytest/test_mitigation.py (hunk 1 above)
Comment (natestemen) [reply frontend/test/pytest/test_mitigation.py:24]: Happy to if catalyst team wants, but otherwise looks like it's not needed.

**Pull request merged**